modules/test/playwright/tests/frontend-editor-ckeditor-web/classic.spec.ts

```diff
@@ -141,3 +141,111 @@ test(
+				widgetName:
+					'com_liferay_editor_ckeditor_sample_web_internal_portlet_CKEditorSamplePortlet',
+			});
+
+			layout = await apiHelpers.headlessDelivery.createSitePage({
+				pageDefinition: getPageDefinition([widgetDefinition]),
+				siteId: site.id,
+				title: getRandomString(),
+			});
+		});
+
+		await test.step('Drag and drop accordion', async () => {
+			await page.goto(
+				`${liferayConfig.environment.baseUrl}/web${site.friendlyUrlPath}${layout.friendlyUrlPath}`
+			);
+
+			const ckeditorSampleWidgetClassicTab = page.getByRole('tab', {
+				name: 'Classic',
+			});
+
+			await ckeditorSampleWidgetClassicTab.waitFor({state: 'visible'});
+			await ckeditorSampleWidgetClassicTab.click();
+
+			const ckeditorEditorBody = page
+				.getByRole('tabpanel', {name: 'Classic'})
+				.frameLocator('iframe[title="editor"]')
+				.getByRole('heading', {name: 'Classic Editor'});
+
+			await ckeditorEditorBody.click();
+
+			await page.keyboard.press('Enter');
+
+			const imageButton = page.getByLabel('Image', {exact: true});
+
+			await imageButton.waitFor({state: 'visible'});
+			await imageButton.click();
+
+			const siteAndLibrariesLink = page
+				.frameLocator('iframe[title="Select Item"]')
+				.getByRole('link', {name: 'Sites and Libraries'});
+
+			await siteAndLibrariesLink.waitFor({state: 'visible'});
+			await siteAndLibrariesLink.click();
+
+			const liferayLink = page
+				.frameLocator('iframe[title="Select Item"]')
+				.getByRole('link', {name: 'Liferay'});
+
+			await liferayLink.waitFor({state: 'visible'});
+			await liferayLink.click();
+
+			const liferayImagesLink = page
+				.frameLocator('iframe[title="Select Item"]')
+				.getByRole('link', {name: 'Provided by Liferay'});
+
+			await liferayImagesLink.waitFor({state: 'visible'});
+			await liferayImagesLink.click();
+
+			const astronautImage = page
+				.frameLocator('iframe[title="Select Item"]')
+				.getByText('astronaut.png');
+
+			await astronautImage.waitFor({state: 'visible'});
+			await astronautImage.click();
+
+			const astronautEditorImage = page
+				.getByRole('application', {name: 'Rich Text Editor,'})
+				.frameLocator('iframe[title="editor"]')
+				.locator('img')
+				.first();
+
+			await astronautEditorImage.waitFor({state: 'visible'});
+			await astronautEditorImage.hover();
+
+			const dragAndDropButton = page
+				.getByRole('application', {name: 'Rich Text Editor,'})
+				.frameLocator('iframe[title="editor"]')
+				.getByTitle('Click and drag to move');
+
+			await dragAndDropButton.dragTo(ckeditorEditorBody);
+
+			const accordionWasDragAndDrop = await page.evaluate(() => {
+				const iframeElement = document.querySelector('iframe');
+
+				const iframeDocument = iframeElement.contentWindow.document;
+				const astronautElement = iframeDocument.querySelector(
+					'img.cke_widget_element'
+				);
+
+				const h1Element = astronautElement.parentElement.parentElement;
+
+				return h1Element.tagName === 'H1';
+			});
```

Comment:
Can we avoid `evaluate` here? Something like 
```js
.frameLocator('iframe').locator('h1 > * > img.cke_widget_element')
```
and check if it is attached.

There is also probably a better way to distinguish between iframes. `document.querySelector('iframe')` just takes first on in DOM, which is fragile. In this PR you are using `'iframe[title="editor"]'` which is fine if it's unique. If it's not, you can pick an attribute that makes it unique. If there is none, you can probably add one in code. Only if you cannot add in code, as a last resort, there is `first()` ot `nth()`.